stake-pool/program/tests/update_stake_pool_balance.rs

```diff
@@ -166,6 +159,87 @@ async fn success() {
     assert_eq!(pool_token_supply, stake_pool.pool_token_supply);
 }
 
+#[tokio::test]
+async fn success_ignoring_extra_lamports() {
+    let (mut context, stake_pool_accounts, stake_accounts) = setup().await;
+
+    let pre_balance = get_validator_list_sum(
+        &mut context.banks_client,
+        &stake_pool_accounts.reserve_stake.pubkey(),
+        &stake_pool_accounts.validator_list.pubkey(),
+    )
+    .await;
+    let stake_pool = get_account(
+        &mut context.banks_client,
+        &stake_pool_accounts.stake_pool.pubkey(),
+    )
+    .await;
+    let stake_pool = try_from_slice_unchecked::<StakePool>(&stake_pool.data.as_slice()).unwrap();
+    assert_eq!(pre_balance, stake_pool.total_stake_lamports);
+
+    let pre_token_supply = get_token_supply(
+        &mut context.banks_client,
+        &stake_pool_accounts.pool_mint.pubkey(),
+    )
+    .await;
+
+    let error = stake_pool_accounts
+        .update_stake_pool_balance(
+            &mut context.banks_client,
+            &context.payer,
+            &context.last_blockhash,
+        )
+        .await;
+    assert!(error.is_none());
```

Comment:
Unfortunately, this isn't exactly the case here, since we're getting back `Option<Error>`.  If it's a success, we'll have `None`, so if we `unwrap()` it'll panic.  I ended up down a bit of a rabbit hole around the concept of `unwrap_none()` which unfortunately isn't going in to the std lib after a lot of discussion.  There's a few weirdnesses in the test code here...  a `Result<(), Error>` would probably be best